tests/schema/test_union.py

```diff
@@ -662,3 +662,30 @@ class Query:
         union: Union[Something, AnnotatedInt]
 
     strawberry.Schema(query=Query)
+
+
+@pytest.mark.parametrize(
+    "annotation", ("int", "str", "float", "list[str]", "List[str]")
+)
+def test_raises_on_union_of_scalars(annotation: str):
+    # using forward refs because they deduced later
+    # catching all edge cases.
+    if annotation == "list[str]" and sys.version_info < (3, 9, 0):
+        return
```

Comment:
I think you can do `pytest.skip()` in here

Reply:
hmm that is just for one parameterization though... You know how to do that?